### ISSUE TITLE
Password/Rofi/Mutt/etc integration

### ISSUE THREAD

**Issue by illusaen** [olzraiti/dotfiles#6]
Hello,

I'm a bit confused as to how your password system works. I see you run decrypt at init - do you decrypt ALL your passwords then? How do you avoid the GTK dialog that gpg pops up if you're running programs at the start?

Thanks much!

**Comment by olzraiti** [comment]
Hey,

The decrypting at init isn't related to the password system - the decrypt script decrypts some files in my dotfiles repo (I want to be able to synchronize them easily across my computers, but I don't want them to be public).

Using the password scripts doesn't require any initialization - just run `paste_pass` eg.

**Comment by illusaen** [comment]
Ahh I see. So looking at your paste_pass (and related) scripts, it seems like you're decrypting and reading from the pass_clipped_fifo file. However, I can't find where you ever write to it. What am I missing? :)

**Comment by olzraiti** [comment]
`clip_pass` outputs the selected pass entry's output to `/tmp/pass_clipped_fifo`.

**Comment by illusaen** [comment]
Oh, I only saw the `pass_picked` part. _whooosh_. One last question - how does the coproc for the rofi `pass.sh` script work?

**Comment by olzraiti** [comment]
Good question! I wrote the rofi `pass.sh` by looking at how the [rofi examples](https://github.com/DaveDavenport/rofi/tree/master/Examples) work and they use coproc for background processes. I actually had to do some reading before answering, and it seems that in this case a regular background process (appending a `&` to the end of the command) would suffice, because there is no need to feed any data to the background process.

EDIT: Actually there is no need for a background process at all... I'll update the script.
EDIT2: Updated. This might actually solve a bug that I haven't been able to figure out :)

**Comment by illusaen** [comment]
Glad to (hopefully) be of help. Thanks so much for your time - I really enjoy your setup!

**Comment by illusaen** [comment]
By the way, how is your password-store set up (directory structure)? I'm having trouble using your find command to actually find the files, much less using the window name :)

**Comment by olzraiti** [comment]
Heh, seems like I broke the system last time. I fixed it and did some improvements.

My password store is simple - directions are one-level deep. I improved the path checking to use regexp though, should be more complete.

I highly recommend checking out [rofi-pass](https://github.com/carnager/rofi-pass) and [autopass](https://github.com/jreinert/autopass) before using these hacky scripts of mine. :)
